test_hyperparameter_clustering.py

```
@@ -0,0 +1,341 @@
+    plt.scatter(embeddings_2d[mask, 0], embeddings_2d[mask, 1], 
+               c=[colors[i]], label=f'Cluster {i+1} ({np.sum(mask)} sentences)', 
+               alpha=0.8, s=50)  # alpha를 0.8로 올리고 점 크기도 키움
+
+plt.title(f'Sentence Clustering Result with Optimal K={recommended_k}', fontsize=16, fontweight='bold')
+plt.xlabel('PCA Component 1', fontsize=12)
+plt.ylabel('PCA Component 2', fontsize=12)
+plt.legend(bbox_to_anchor=(1.05, 1), loc='upper left', fontsize=10)
+plt.grid(True, alpha=0.3)
+plt.tight_layout()
+plt.savefig('optimal_clustering_result.png', dpi=150, bbox_inches='tight')
+print(f"최종 군집화 결과가 'optimal_clustering_result.png'로 저장되었습니다.")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)